### ISSUE TITLE
zBots jerky appearance from Non Steam client only

### ISSUE THREAD

**Issue by yunodeb** [rehlds/ReGameDLL_CS#1026]
Hello,

There is an issue with zBot, if youre playing with a Non Steam client, they have a jerky appearance.
I've already tried several Non Steam client, but without result.

I don't have this issue with a Steam client.

I don't have this issue with a Non Steam client and PODBot.

Here is my server config :

Protocol version 48
Exe version 1.1.2.7/Stdio (cstrike)
ReHLDS version: 3.13.0.788-dev
Build date: 07:36:33 Jul 12 2023 (3378)
Build from: https://github.com/dreamstalker/rehlds/commit/f955b07

ReGameDLL version: 5.26.0.668-dev
Build date: 19:02:48 Dec 31 2023
Build from: https://github.com/s1lentq/ReGameDLL_CS/commit/d3c0ec8

Metamod-r v1.3.0.138, API (5:13)
Metamod-r build: 11:50:32 Nov 28 2023
Metamod-r from: https://github.com/theAsmodai/metamod-r/commit/5179c25

Currently loaded plugins:
      description      stat pend  file                      vers             src  load  unload
 [ 1] SafeNameAndChat  RUN   -    SafeNameAndChat.so        v1.1             ini  ANY   ANY
 [ 2] Reunion          RUN   -    reunion_mm_i386.so        v0.1.0.137       ini  Start Never
 [ 3] ReAuthCheck      RUN   -    reauthcheck_mm_i386.so    v0.1.6           ini  Start Never
 [ 4] VoiceTranscoder  RUN   -    VoiceTranscoder.so        v2017RC5         ini  ANY   ANY
 [ 5] AMX Mod X        RUN   -    amxmodx_mm_i386.so        v1.10.0.5467     ini  Start ANY
 [ 6] WHBlocker        RUN   -    whblocker_mm_i386.so      v1.5.697         ini  Chlvl ANY
 [ 7] Rechecker        RUN   -    rechecker_mm_i386.so      v2.7             ini  Chlvl ANY
 [ 8] ReSRDetector     RUN   -    resrdetector_mm_i386.so   v0.1.0           ini  Chlvl ANY
 [ 9] ReSemiclip       RUN   -    resemiclip_mm_i386.so     v2.3.9           ini  Chlvl ANY
 [10] HitBox Fix       RUN   -    hitbox_fix_mm_i386.so     v1.1.4           ini  Start ANY
 [11] Fun              RUN   -    fun_amxx_i386.so          v1.10.0.5467     pl5  ANY   ANY
 [12] Engine           RUN   -    engine_amxx_i386.so       v1.10.0.5467     pl5  ANY   ANY
 [13] FakeMeta         RUN   -    fakemeta_amxx_i386.so     v1.10.0.5467     pl5  ANY   ANY
 [14] GeoIP            RUN   -    geoip_amxx_i386.so        v1.10.0.5467     pl5  ANY   ANY
 [15] CStrike          RUN   -    cstrike_amxx_i386.so      v1.10.0.5467     pl5  ANY   ANY
 [16] CSX              RUN   -    csx_amxx_i386.so          v1.10.0.5467     pl5  ANY   ANY
 [17] Ham Sandwich     RUN   -    hamsandwich_amxx_i386.so  v1.10.0.5467     pl5  ANY   ANY
 [18] ReAPI            RUN   -    reapi_amxx_i386.so        v5.24.0.300-dev  pl5  ANY   Never
18 plugins, 18 running

AMX Mod X 1.10.0.5467 (http://www.amxmodx.org)
Authors:
        David "BAILOPAN" Anderson, Pavol "PM OnoTo" Marko
        Felix "SniperBeamer" Geyer, Jonny "Got His Gun" Bergstrom
        Lukasz "SidLuke" Wlasinski, Christian "Basic-Master" Hammacher
        Borja "faluco" Ferrer, Scott "DS" Ehlert
Compiled: Jan 21 2023 20:27:33
Built from: https://github.com/alliedmodders/amxmodx/commit/27f451a
Build ID: 5467:27f451a
Core mode: JIT+ASM32

Currently loaded plugins:
       id  name                    version     author            url                              file         status
 [  1] 0   Admin Base              1.10.0.546  AMXX Dev Team                                      admin.amxx   running
 [  2] 1   Admin Commands          1.10.0.546  AMXX Dev Team                                      admincmd.am  running
 [  3] 2   Admin Help              1.10.0.546  AMXX Dev Team                                      adminhelp.a  running
 [  4] 3   Slots Reservation       1.10.0.546  AMXX Dev Team                                      adminslots.  running
 [  5] 4   Multi-Lingual System    1.10.0.546  AMXX Dev Team                                      multilingua  running
 [  6] 5   Menus Front-End         1.10.0.546  AMXX Dev Team                                      menufront.a  running
 [  7] 6   Commands Menu           1.10.0.546  AMXX Dev Team                                      cmdmenu.amx  running
 [  8] 7   Players Menu            1.10.0.546  AMXX Dev Team                                      plmenu.amxx  running
 [  9] 8   Maps Menu               1.10.0.546  AMXX Dev Team                                      mapsmenu.am  running
 [ 10] 9   Plugin Menu             1.10.0.546  AMXX Dev Team                                      pluginmenu.  running
 [ 11] 10  Admin Chat              1.10.0.546  AMXX Dev Team                                      adminchat.a  running
 [ 12] 11  Anti Flood              1.10.0.546  AMXX Dev Team                                      antiflood.a  running
 [ 13] 12  Scrolling Message       1.10.0.546  AMXX Dev Team                                      scrollmsg.a  running
 [ 14] 13  Info. Messages          1.10.0.546  AMXX Dev Team                                      imessage.am  running
 [ 15] 14  Admin Votes             1.10.0.546  AMXX Dev Team                                      adminvote.a  running
 [ 16] 15  NextMap                 1.10.0.546  AMXX Dev Team                                      nextmap.amx  running
 [ 17] 16  Nextmap Chooser         1.10.0.546  AMXX Dev Team                                      mapchooser.  running
 [ 18] 17  TimeLeft                1.10.0.546  AMXX Dev Team                                      timeleft.am  running
 [ 19] 18  Pause Plugins           1.10.0.546  AMXX Dev Team                                      pausecfg.am  running
 [ 20] 19  Stats Configuration     1.10.0.546  AMXX Dev Team                                      statscfg.am  running
 [ 21] 20  StatsX                  1.10.0.546  AMXX Dev Team                                      statsx.amxx  running
 [ 22] 21  CS Misc. Stats          1.10.0.546  AMXX Dev Team                                      miscstats.a  running
 [ 23] 22  CS Stats Logging        1.10.0.546  AMXX Dev Team                                      stats_loggi  running
 [ 24] 23  POD-Bot Menu            1.3         g4s|figurE.09     unknown                          amxx_podbot  running
24 plugins, 24 running

Currently loaded modules:
      name                    version     author               status
 [ 1] Fun                     1.10.0.546  AMX Mod X Dev Team   running
 [ 2] Engine                  1.10.0.546  AMX Mod X Dev Team   running
 [ 3] FakeMeta                1.10.0.546  AMX Mod X Dev Team   running
 [ 4] GeoIP                   1.10.0.546  AMX Mod X Dev Team   running
 [ 5] nVault                  1.10.0.546  AMX Mod X Dev Team   running
 [ 6] CStrike                 1.10.0.546  AMX Mod X Dev Team   running
 [ 7] CSX                     1.10.0.546  AMX Mod X Dev Team   running
 [ 8] Ham Sandwich            1.10.0.546  AMX Mod X Dev Team   running
 [ 9] ReAPI                   5.24.0.300  Asmodai & s1lent     running
 [10] Sockets                 1.10.0.546  AMX Mod X Dev Team   running
 [11] RegEx                   1.10.0.546  AMX Mod X Dev Team   running
11 modules, 11 correct

Just to let you know @Vaqtincha if there is something to do :)

https://github.com/s1lentq/ReGameDLL_CS/issues/619#issue-872724961 i think this is the same issue here

See you

**Comment by Vaqtincha** [comment]
@franckdeb try change https://github.com/s1lentq/ReGameDLL_CS/blob/master/regamedll/game_shared/bot/bot.h#L34

**Comment by sweetcarolinahoney** [comment]
@franckdeb Yeah, you have to compile it yourself. Changing g_flBotCommandInterval to "1.0 / 60.0" will work just fine.

**Comment by Nord1cWarr1or** [comment]
Explained in the thread — zBots issue mental commands on a fixed interval defined by `g_flBotCommandInterval` (`bot.h:34`), so their position updates look jerky on clients whose interpolation settings differ (common on Non-Steam builds; PODBot behaves differently). Raising `g_flBotCommandInterval` to 1/60 in a custom build smooths it — changing the compiled-in default is an upstream decision. Closing as completed (answered with the exact lever, ref #619).
